### PR TITLE
feat(web): Pension calculator

### PR DESCRIPTION
**Pull request by RunarVestmann**
# Pension calculator

## What

* Bringing https://www.tr.is/reiknivel/ to Ísland.is
* "Custom Page" content type added (so the page can fetch translations, configurations and display an alert banner)
* Custom Page content type gets it's translations from the Translation Namespace content type so I added support for that similar to how other apps in the monorepo do it
* GraphQL endpoint added for the pension calculation (I also added stefna as a code owner to the api domain and the client lib since now we own code there)
* There were multiple copies of a `formatCurrency` function throughout the web project so I reused a single one instead
* `size` prop was added to SelectController component so we can appropriately change the size

## Screenshots / Gifs

https://github.com/island-is/island.is/assets/43557895/a0a3a67e-075f-4a4a-b48c-23be7e8d2579

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by AndesKrrrrrrrrrrr — APPROVED** [review]
CODEOWNERS LGTM

**Review by Toti91 — APPROVED** [review]
Code owned lgtm

File: libs/api/domains/social-insurance/src/lib/dtos/pensionCalculation.input.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  livingCondition?: LivingCondition
+
+  @Field(() => Boolean, { nullable: true })
+  mobilityImpairment?: boolean
+
+  @Field(() => String, { nullable: true })
+  startDate?: string
+
+  @Field(() => Number, { nullable: true })
+  childCount?: number
```
Comment (thorkellmani) [inline libs/api/domains/social-insurance/src/lib/dtos/pensionCalculation.input.ts:60]: FYI you don't need to explicitly declare the Boolean or the String. However, you need to assign the number (`childCount`) to either Int or Float, according to the[ NestJS docs](https://docs.nestjs.com/graphql/resolvers)

`The type function is required when there's the potential for ambiguity between the TypeScript type system and the GraphQL type system. Specifically: it is not required for string and boolean types; it is required for number (which must be mapped to either a GraphQL Int or Float).`

File: libs/api/domains/social-insurance/src/lib/models/pensionCalculation.model.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { Field, ObjectType } from '@nestjs/graphql'
+import { CacheField } from '@island.is/nest/graphql'
+
+@ObjectType('SocialInsurancePensionCalculationResponseItem')
+class PensionCalculationItem {
+  @Field(() => String, { nullable: true })
+  name?: string | null
```
Comment (thorkellmani) [inline libs/api/domains/social-insurance/src/lib/models/pensionCalculation.model.ts:7]: Isn't it unnecessary to have it be both `null` and `nullable`?

File: libs/api/domains/social-insurance/src/lib/utils.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    privatePensionPayments: input.privatePensionPayments,
+    taxCard: input.taxCard,
+    ageOfFirst75DisabilityAssessment: input.ageOfFirst75DisabilityAssessment,
+    installmentClaims: input.installmentClaims,
+    livingConditionRatio: input.livingConditionRatio,
+
+    // Fields that are calculated or mapped
+    hurryPension,
+    delayPension,
+    start: startPension, // TODO: perhaps this is wrong?
```
Comment (thorkellmani) [inline libs/api/domains/social-insurance/src/lib/utils.ts:99]: Todo? :eyes:

File: libs/api/domains/social-insurance/src/lib/utils.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+  }
+
+  let hurryPension = 0
+  let delayPension = 0
+  let startPension: 1 | 2 = 2
+
+  if (input.birthdate) {
+    const birthdate = new Date(input.birthdate)
+
+    const defaultPensionAge = 67 // TODO: hardcoding 67 is not ideal
```
Comment (thorkellmani) [inline libs/api/domains/social-insurance/src/lib/utils.ts:53]: Todo? 👀

File: libs/api/domains/social-insurance/src/lib/models/pensionCalculation.model.ts (hunk 2 above)
Comment (RunarVestmann) [reply libs/api/domains/social-insurance/src/lib/models/pensionCalculation.model.ts:7]: I want to keep it though 
image

**Review by mannipje — APPROVED** [review]

**Review by thorkellmani — APPROVED** [review]
LGTM 🦆

**Pull request merged**